mne/simulation/source.py

```diff
@@ -267,6 +267,13 @@ def simulate_stc(src, labels, stc_data, tmin, tstep, value_fun=None,
         hemi_ind = hemi_to_ind[label.hemi]
         src_sel = np.intersect1d(src[hemi_ind]['vertno'],
                                  label.vertices)
+        if len(src_sel) == 0:
+            idx = np.searchsorted(src[hemi_ind]['vertno'],
+                                  label.vertices)
+            src_len = len(src[hemi_ind]['vertno']) - 1
+            idx[idx > src_len] = src_len
+            src_sel = np.unique(src[hemi_ind]['vertno'][idx])
+
```

Comment:
when I see `src_sel` I read: selected sources, or sorces that are {selected, active}. This makes me think of a list of some sort (in this case represented by a numpy array).

More over, the block of code in this PR shows that this is the case. 'cos it reads: `if the list of selected sources is empty do something`. What breaks all this logic is that then you reuse this very same variable to store a scalar, perform a selection use it back as a recipient of the computation and then there is no else statement.

I would recommend you to create a function that would make everything much more readable (choose `foo`, `bar` and `foobar` approperly):

```py
src_sel = _select_foobar(foo=src[hemi_ind]['vertno'], bar=label.vertices)
```

Then when you write the function I would use:
```py
def _select_foobar(foo, bar):
    intersection = np.intersect1d(src[hemi_ind]['vertno'], labels.vertices)
    if intersection.size > 0:  # This is how np recomends to check for not empty
         return intersection
    else:
         return something else
```

Reply:
Still, the snipped its complicated to read. 

`src_sel` it ends up being `[54]` which is the same as `src[hemi_ind]['vertno'][0]`

I'm confused.

Reply:
> src_sel it ends up being [54] which is the same as src[hemi_ind]['vertno'][0]

that's what it should be in test.

My label contains a single vertex with index 0:
`i = np.where(fwd['src'][0]['inuse'] == 0)[0][0]`

This vertex is not in the source space. So the closest vertex index from the source space to label vertex is  `[54] `

Reply:
factoring in a function and creating toy examples to fully tested in a isolated enviroment would provide better understanding